When writing RFC2317 zones, convert "/" to "-" in the filename.

sim0nx opened the pull request:
When trying to write RFC2317 zones, of the form e.g. *0/25.10.10.10.in-addr.arpa.*, octodns-sync fails with:
```
...
with open(filename, 'w') as fh:
^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: './test/0/25.10.10.10.in-addr.arpa.'
```

To avoid this, we replace the **/** with a **-** in the filename.

ross reviewed (commented):
Good catch and the change makes sense. Main request would be to add a test of the new functionality so that we can make sure that it works as expected now and doesn't get broken in the future. 

It'd probably look a lot like https://github.com/octodns/octodns-bind/blob/dffed120c095009c252ac7f6916c3b2d26026480/tests/test_provider_octodns_bind.py#L218-L246, just swapping the `unit.tests.` zone name for something with a `/`.

sim0nx commented:
> Good catch and the change makes sense. Main request would be to add a test of the new functionality so that we can make sure that it works as expected now and doesn't get broken in the future.
> 
> It'd probably look a lot like
> 
> https://github.com/octodns/octodns-bind/blob/dffed120c095009c252ac7f6916c3b2d26026480/tests/test_provider_octodns_bind.py#L218-L246
> 
> , just swapping the `unit.tests.` zone name for something with a `/`.

Thanks for your feedback.
I added a test-case inspired by *test_apply* as you suggested.
I had to add an NS record as else it would auto-generate one for the SOA record based on the zone name which is invalid. Same goes for the hostmaster_email.

After asserting that the written output is as expected, I stripped the rest of the *test_apply* part as that is IMHO not required to test again and was problematic with the NS I added earlier (duplicate records error..).

I am not too familiar with the code base so I hope it is fine like that :-)

ross commented:
Thanks
